src/ui/components/Library/Team/View/Tests/TestListItem.tsx

```diff
@@ -1,20 +1,44 @@
-import { useContext } from "react";
+import { useContext, useMemo } from "react";
 
 import { Test } from "shared/test-suites/TestRun";
 import HighlightedText from "ui/components/Library/Team/View/TestRuns/HighlightedText";
 import Icon from "ui/components/shared/Icon";
 
 import { TestContext } from "./TestContextRoot";
-import styles from "../../../Library.module.css";
+import styles from "../NewTestRuns/TestRuns.module.css";
 
-function Status({ failureRate }: { failureRate: number }) {
-  const status = failureRate > 0 ? "fail" : "success";
-  const displayedFailureRate = Number((failureRate * 100).toFixed(0));
+function Status({ test }: { test: Test }) {
+  const { sortBy } = useContext(TestContext);
 
-  if (failureRate > 0) {
+  // We are showing rate in the pill depending on the sortBy.
+  // If sortBy is failureRate or alphabetical, we show rate in this order of priority: failureRate, flakyRate, success.
+  // Otherwise with sortBy flakyRate, we show rate in this order: flakyRate, failureRate, success.
```

Comment:
Here's how we decide which rate to show.

Reply:
It feels unexpected to me to be sorting by flaky rate but sometimes show the failure rate. I'd favor always showing the flaky rate when that's the sort order even if the rate is 0 and there were some failures. My assumption here is the use case for this view is to drive down flake so bringing focus to failures (but not flakes) would be distracting.

Reply:
It'll now show either flaky or failure rate for all tests. Flaky when flaky rate is selected. Failure when either alphabetical or failure rate is selected. Does this seem okay to you?

Reply:
Feels bit odd when in flaky sort, we'll never show a green check. 🤔 But maybe that only makes sense in failure rate sort.